dttools/src/mkdir_recursive.h

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright (C) 2015- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#ifndef MKDIR_RECURSIVE_H
+#define MKDIR_RECURSIVE_H
+
+#include "compat-at.h"
+
+/** @file mkdir_recursive.h Create a new directory recursively. */
+
+/** Create a new directory recursively.
+@param fd The directory path is relative to.
+@param path The full path of a directory.  It is not necessary for all components of the path to exist.
+@param mode The desired unix mode bits of the directory and parents.
+@return -1 on failure, 0 on success.
+*/
+
+int mkdirat_recursive(int fd, const char *path, mode_t mode);
+#define mkdir_recursive(path,mode) mkdirat_recursive(AT_FDCWD,path,mode)
```

Comment:
Avoid macros unless there is a really solid reason.
Make these real functions that invoke the underlying ones.